main.go

```diff
@@ -413,6 +450,51 @@ func convert(req *plugin.CodeGeneratorRequest) (*plugin.CodeGeneratorResponse, e
 	return res, nil
 }
 
+func addExtensions(msg *descriptor.DescriptorProto, extensions []*descriptor.FieldDescriptorProto, pkg string) {
+	for _, ext := range extensions {
+		extendeeNodes := strings.Split(ext.GetExtendee(), ".")
+		pkgNodes := strings.Split(pkg, ".")
+
+		// We already know this extendee cannot be in this package.
+		if len(pkgNodes) > len(extendeeNodes) {
+			continue
+		}
+
+		// skipping leading "."
+		nodeIdx := 1
+		for _, node := range pkgNodes {
+			if node != extendeeNodes[nodeIdx] {
+				nodeIdx = 0
+				break
+			}
+			nodeIdx += 1
+		}
+
+		// Not the right package or message
+		if nodeIdx == 0 || msg.GetName() != extendeeNodes[nodeIdx] {
```

Comment:
Explicit `mismatch` for clarity, feel free to rename

```suggestion
		nodeIdx := 1
		mismatch := false
		for _, node := range pkgNodes {
			if node != extendeeNodes[nodeIdx] {
				mismatch = true
				break
			}
			nodeIdx += 1
		}

		// Not the right package or message
		if mismatch || msg.GetName() != extendeeNodes[nodeIdx] {
```